rfcs/text/0009_screenshot_mode_service.md

```diff
@@ -0,0 +1,260 @@
+loading UI components into the browser, and help the application make
+case-by-case bases for loading components and the parameters that get passed
+into the components.
+
+In most cases, the information coming from this service would help the
+application which UI components should **not** be loaded, as applications will
+use the service response for skipping components that aren't needed for
+presenting the data in a screenshot, and thus work towards better performance
+for the display-only purpose.
+
+More background on how Reporting currently works, including the lifecycle of
+creating a PNG report, is here: https://github.com/elastic/kibana/issues/59396
+
+# Basic example
+
+When Kibana loads initially, there is a Newsfeed component in the UI that
+checks internally cached records to see if it must fetch the Elastic News
+Service for newer items. When the Screenshot Mode Service is implemented, the
+Newsfeed component has a source of information to check on whether or not it
+should load in the Kibana UI. If it can avoid loading, it avoids an unecessary
+HTTP round trip, which weigh heavily on performance.
+
+# Motivation
+
+Kibana Reporting is a commercial feature in Elastic and is highly capable of
+loading the application pages and converting them into a screenshot for PNG or
+PDF export. However, the way it works has downsides with performance,
+maintainability, and expanding it as a tool that can power higher-level
+features. The solution to those downsides is to have Kibana pages themselves
+become more capable at presenting themselves for screenshot capture reports.
+With the Screenshot Mode Service available in core, Reporting could drop the
+task that it currently has which hurt performance: wasted rendering that is
+replaced with custom styles that make the page "reportable."
+
+The technical advantage of having such as service also leads to making Kibana
+application pages "printable", in the sense that sending the page to a printer
+for a hard copy results in something more meaningful and specialized for paper
+than today's Kibana can guarantee. This isn't a big concern for Kibana since
+there isn't the expectation to improve printing Kibana, but that technical
+direction is appropriate for improving PDF report generation.
+
+# Detailed design
+
+The Screenshot Mode Service is a callable API availble in the context of a
+request on the server side, and as a pubic coreSetup Javascript API.
+
+The data provided by the Screenshot Mode are signals about:
+ - whether or not the context of the page load is for a screenshot capture
+ - layout and dimensions that the result image is expected to be
+ - time zone for which to format the page data
+
+There are a few ways for the service to obtain the data for its internal state.
+The most typical way would be to have a URL query string variable with the
+raw data and have the service read it when an API client is constructed. The
```

Comment:
By "API client", are you referring to an instance of `ScreenshotModeServiceApi` described below?

Reply:
Pretty much. I reworded this paragraph: c33de1b2c7

Instead of "when the API client is constructed" I think I mean "in the `setup` method of the plugin" where it reads the URL and forms its interface data.